common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "@fluidframework/common-definitions";
+
+
+interface Task {
+    handle: () => any,
+}
+
+
+/**
+ * Helper class that is used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ */
+export default class IdleTaskScheduler {
+    //List of tasks waiting to be run
+    private idleTaskList:  Task[] = [];
+    // A reference to the task currently being processed.
+    private taskHandle = 0;
```

Comment:
We should not assume any type here or any default values (other than undefined).
Please use

private taskHandle: ReturnType<typeof requestIdleCallback> | undefined = undefined;